api/http/handler/endpoints/endpoint_list.go

```diff
@@ -51,7 +51,15 @@ func (handler *Handler) endpointList(w http.ResponseWriter, r *http.Request) *ht
 	}
 
 	if search != "" {
-		filteredEndpoints = filterEndpointsBySearchCriteria(filteredEndpoints, endpointGroups, search)
+		tags, err := handler.TagsService.Tags()
+		if err != nil {
+			return &httperror.HandlerError{http.StatusInternalServerError, "Unable to retrieve tags from the database", err}
+		}
+		tagsMap := make(map[portainer.TagID]portainer.Tag)
```

Comment:
What about using a `map[portainer.TagID]string` instead? No need to reference the entire tag since we're only using the name for search/comparison.

Reply:
:+1: